A volume is not set properly for cockroachdb/builder docker image

`builder.sh` and `build-docker-deploy.sh` do not set `/go/src/github.com/cockroachdb/cockroach` volume properly for `cockroachdb/builder` docker image if they are not launched from `cockroach` directory. The following text shows the effect if they are launched from `cockroach/build` directory.

In `cockroach` directory, run the following commands to launch `build-docker-deploy.sh` :

```
> cd build
> ./build-docker-deploy.sh 
make: *** No rule to make target 'release'.  Stop.
```

If `builder.sh` is launched from `cockroach/build` folder, `/go/src/github.com/cockroachdb/cockroach` will be mapped to `cockroach/build` directory instead of `cockroach` directory.

--- a/build/build-docker-deploy.sh
+++ b/build/build-docker-deploy.sh
@@ -23,8 +23,11 @@ if [ "${1-}" = "docker" ]; then
     exit 0
 fi
 
+# Relative path from the top-level cockroach directory to the current script.
+relative_path_to_self="$(basename $(cd $(dirname $0); pwd))/$(basename $0)"
+
 # Build the cockroach and test binaries.
-$(dirname $0)/builder.sh $0 docker
+$(dirname $0)/builder.sh ${relative_path_to_self} docker
 
 # Build the image.
 docker build --tag=cockroachdb/cockroach "$(dirname $0)/deploy"

--- a/build/builder.sh
+++ b/build/builder.sh
@@ -44,6 +44,9 @@ fi
 buildcache_dir="buildcache"
 uicache_dir="uicache"
 
+# Absolute path to the toplevel cockroach directory.
+cockroach_toplevel="$(dirname $(cd $(dirname $0); pwd))"
+
 # Run our build container with a set of volumes mounted that will
 # allow the container to store persistent build data on the host
 # computer.
@@ -69,7 +72,7 @@ docker run -i ${tty-} ${rm} \
   --volume="${gopath0}/bin/linux_amd64:/go/bin" \
   --volume="${HOME}/${buildcache_dir}:/${buildcache_dir}" \
   --volume="${HOME}/${uicache_dir}:/${uicache_dir}" \
-  --volume="${PWD}:/go/src/github.com/cockroachdb/cockroach" \
+  --volume="${cockroach_toplevel}:/go/src/github.com/cockroachdb/cockroach" \
   --workdir="/go/src/github.com/cockroachdb/cockroach" \
   --env="CACHE=/${buildcache_dir}" \
   --env="PAGER=cat" \

